README.md

```diff
@@ -244,39 +244,44 @@ You can also add them to the end of `~/.profile` to set this by default in all s
 
 #### MacOS dependencies installation
 
-To install GCC and CMake on MacOS we recommend to use [Homebrew](https://brew.sh/) package manager. With Homebrew
+Install Xcode Command Line Tools if you don't have them. Run:
+```sh
+xcode-select --install
+```
+Follow the prompts to contunue.
+
+To install GCC and CMake on macOS we recommend to use [Homebrew](https://brew.sh/) package manager. With Homebrew
 installed, run the following commands:
 ```sh
-brew install gcc cmake 
+brew install gcc@14 cmake 
 ```
 After installation, check `cmake --version`. If command is not found, then you need to add to environment path to
 homebrew installed packages. To do this open `~/.zprofile` (for Zsh) or
 `~/.bash_profile` (for Bash) and add to the end of the file the output of `brew shellenv`.
 After that, restart the terminal and check the version of CMake again, now it should be displayed.
 
-Then, check the installed version of GCC:`brew info gcc`. You must see something like `==> gcc: stable X.Y.Z ...`. 
-Check that `gcc-X` and `g++-X` commands work, where X is the version from this output 
-(this designation continues to be used further).
-
-After you need to install Boost library. Please, don't use Homebrew for this, as it may not work correctly.
-Instead, download the latest version of Boost from the [official website](https://www.boost.org/users/download/).
-After downloading, unpack the archive to the `/usr/local` directory or another directory of your
-choice.
+Then you need to install Boost library built with GCC. Please avoid using Homebrew for this, as the Boost version provided by Homebrew 
+is built with Clang, which has a different ABI. Instead, download the latest version of Boost from the [official website](https://www.boost.org/users/download/).
+After downloading, unpack the archive to the `/usr/local` directory or another directory of your choice.
```

Comment:
Since we are providing a very detailed guide, then perhaps we can also provide commands for downloading the boost (for example, use the current latest version of boost and the path `/usr/local` in the commands).

I would say that I would be inclined to do so, but I would not insist.